app/electron/main.ts

```diff
@@ -644,6 +647,62 @@ function startElecron() {
       }
     });
 
+    /**
+     * Data sent from the renderer process when a 'run-command' event is emitted.
+     */
+    interface CommandData {
+      /** The unique ID of the command. */
+      id: string;
+      /** The command to run. */
+      command: string;
+      /** The arguments to pass to the command. */
+      args: string[];
+      /**
+       * Options to pass to the command.
+       * See https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options
+       */
+      options: {};
+    }
+
+    /**
+     * Handles 'run-command' events from the renderer process.
+     *
+     * Spawns the requested command and sends 'command-stdout',
+     * 'command-stderr', and 'command-exit' events back to the renderer
+     * process with the command's output and exit code.
+     *
+     * @param event - The event object.
+     * @param eventData - The data sent from the renderer process.
+     */
+    function handleRunCommand(event: IpcMainEvent, eventData: CommandData): void {
+      // Only allow "minikube", "az", and "ls" commands
+      const validCommands = ['minikube', 'az', 'ls'];
+      if (!validCommands.includes(eventData.command)) {
+        console.log(`Ignoring invalid command: ${eventData.command}`);
+        return;
+      }
+
+      const child: ChildProcessWithoutNullStreams = spawn(
+        eventData.command,
+        eventData.args,
+        eventData.options
+      );
+
+      child.stdout.on('data', data => {
```

Comment:
Instead of `any` type for data, how about `(data: Buffer | string)` and for below 2 as well?

Reply:
done